README.md

```diff
@@ -1,2 +1,66 @@
 # fluent-siren-client
 Fluent client for Siren Hyper Media APIs
+
+Fetches a [Siren](https://github.com/kevinswiber/siren) object using the provided href and request function, and parses 
+it into a [node-siren-parser](https://github.com/Brightspace/node-siren-parser) Entity. 
+Configures the Entity in such a way that uri links can be followed, and actions can be performed. Result of 
+following a link is another fluent Entity.
+
+## Installation
+
+Install from NPM:
+```shell
+npm install fluent-siren-client
+```
+
+## Usage
+
+```javascript
+describe('SirenClient', () => {
+  const SirenClient = require('fluent-siren-client');
+
+  function requestFn(href, method, fieldValues) {
+    if (href === 'http://external.website.com') {
+      return {
+        body: { properties: { name: 'external' } },
+        contentType: 'application/vnd.siren+json',
+      };
+    }
+    if (href === 'http://api.example.com/entity') {
+      return {
+        body: { properties: { name: 'entity' } },
+        contentType: 'application/vnd.siren+json',
+      };
+    }
+    if (href === 'http://api.example.com/action') {
+      return {
+        body: { properties: { name: 'action' } },
+        contentType: 'application/vnd.siren+json',
+      };
+    }
+    if (href === 'http://api.example.com') {
+      return {
+        body: {
+          properties: { name: 'api' },
+          links: [{ rel: ['external'], href: 'http://external.website.com' }],
+          entities: [{
+            rel: ['item'],
+            links: [{ rel: ['self'], href: 'http://api.example.com/entity' }],
+          }],
+          actions: [{ name: 'action', href: 'http://api.example.com/action' }],
+        },
+        contentType: 'application/vnd.siren+json',
+      };
+    }
+    return {};
+  }
```

Comment:
I don't think we need any of this extra context for the readme.
Maybe we could have example of how to setup the client - but the first thing should just be what is contained on lines 59-63

Then maybe followed by setup.
People generally want to see the main use case up front then details after